server/src/main/java/org/elasticsearch/cluster/metadata/DesiredNode.java

```diff
@@ -95,6 +107,26 @@ private static Version parseVersion(String version) {
         return Version.fromString(version);
     }
 
+    public enum MembershipStatus {
+        UNKNOWN((short) 0),
```

Comment:
I wonder if UNKNOWN should be "NEW" instead? It is a new node that we have not seen yet.